lib/chargebee/rest.rb

```diff
@@ -55,7 +55,12 @@ def self.request(method, url, env, params=nil, headers={})
             raise IOError.new("IO Exception when trying to connect to chargebee with url #{opts[:url]} . Reason #{e}",e)        
       end
       rheaders = response.headers
+
       rbody = response.body
+      if headers.keys.any? { |k| k.downcase == 'accept-encoding' } && rheaders[:content_encoding] == 'gzip' && rbody && !rbody.empty?
+          rbody = Zlib::GzipReader.new(StringIO.new(rbody)).read
+      end
```

Comment:
Manually decode the gzipped response if `Accept-Encoding` header was manually set and response containts `Content-Encoding: gzip`.